File: spot-it.c
#include <stdio.h>

#define P 7  // Prime number for grid
#define S 8  // Symbols per card
#define C 57 // Total cards

int main() {
    int cards[C][S] = {0};
    
    for (int i = 0; i < S; i++) cards[0][i] = i + 1;    // First card: 1 to 8
    
    for (int i = 0; i < P; i++)                         // Grid cards
        for (int j = 0; j < P; j++) {
            int* card = cards[i * P + j + 1];
            card[0] = i + 2;
            for (int k = 0; k < P; k++)
                card[k + 1] = ((i * k + j) % P) + P * k + P + 2; }

    for (int i = 0; i < P; i++) {                       // Infinity cards
        int* card = cards[P * P + 1 + i];
        card[0] = 1;
        for (int j = 0; j < P; j++)
            card[j + 1] = P + 2 + j + i * P; }

    for (int i = 0; i < C; i++) {                       // Print cards
        printf("Card %2d:", i + 1);
        for (int j = 0; j < S; j++) printf(" %2d", cards[i][j]); 
        printf("\n"); }
}

File: spot-it.py
P, S, C = 7, 8, 57                  # Prime, Symbols per card, Total cards
cards = [[0]*S for _ in range(C)]
cards[0] = [i+1 for i in range(S)]  # First card

for i in range(P):                  # Grid cards
    for j in range(P):
        cards[i*P+j+1] = [i+2] + [((i*k+j)%P)+P*k+P+2 for k in range(P)]

for i in range(P):                  # Infinity cards
    cards[P*P+1+i] = [1] + [P+2+j+i*P for j in range(P)]

for i, card in enumerate(cards):    # Print cards
    print(f"Card {i+1:2}:", *[f"{n:2}" for n in card])
